[B-enguin] iP

B-enguin opened the pull request:
## Duke Pro

> “Your mind is for having ideas, not holding them.” – David Allen
>  ([source](https://dansilvestre.com/productivity-quotes))

- text-based
* easy to learn
+ ~FAST~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (comming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

    public class Main {
        public static void main(String[] args) {
            Application.launch(MainApp.class, args);
        }
    }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

B-enguin committed:
Implemented Level-0 Requirements

B-enguin committed:
Implemented Level-1 Requirements

B-enguin committed:
Implemented Level-2 Requirements

B-enguin committed:
Implemented Level-3, A-Classes Requirements

B-enguin committed:
Implemented Level-4, A-Inheritance Requirements

B-enguin committed:
Implemented A-TextUiTesting Requirements

B-enguin committed:
Implemented Level-5, A-Exceptions Requirements

B-enguin committed:
Implemented Level-6, A-Collections Requirements

B-enguin committed:
Implemented A-Enums Requirements

B-enguin committed:
Fix A-TextUiTesting to account for A-Enum Escape Codes

B-enguin committed:
Refactor for A-MoreOOP, A-Packages. Added requirements for Level-7

B-enguin committed:
Merge branch 'branch-Level-7'

B-enguin committed:
Added Tracking for save data

B-enguin committed:
Implemented Level-8 Requirements

B-enguin committed:
Merge branch 'branch-Level-8'

B-enguin committed:
Merge remote-tracking branch 'origin/add-gradle-support'

B-enguin committed:
Fix build.gradle for project structure

B-enguin committed:
Add A-JUnit requirements

B-enguin committed:
Merge branch 'branch-A-JUnit'

B-enguin committed:
Add Javadocs

B-enguin committed:
Add A-CodingStandard Requirements

B-enguin committed:
Add Level-9 Requirements

B-enguin committed:
Merge branch 'branch-Level-9'

B-enguin committed:
Add Level-9 Requirements

B-enguin committed:
Fix missing getDescription from Task

B-enguin committed:
Add A-Checkstyle Requirements.

Add support for checkstyle,
Modify project files to follow cource checkstyle requirments

B-enguin committed:
Merge branch 'branch-A-Checkstyle'

B-enguin committed:
Modify project to follow checkstyle.

B-enguin committed:
Merge branch 'branch-A-Checkstyle'

B-enguin committed:
Add support for GUI.

Initial files required for GUI added

B-enguin committed:
Replace interface with GUI.

Partially migrated to using GUI over text-based interface, in partial fulfilment of Level-10 requirements.
Removed isExit() from all classes which inherit Command.java

B-enguin committed:
Add Level-10 Requirements.

Full migration to GUI.

B-enguin committed:
Merge branch 'branch-Level-10'

Cikguseven commented on src/main/java/duke/command/NumberedChoiceCommand.java:
```diff
@@ -0,0 +1,27 @@
+     * Validate arguments to this command.
+     *
+     * @param arguments arguments to validate
+     * @throws DukeException if arguments are invalid
```
Perhaps you could add punctuation behind each parameter description to adhere to the coding standard.

Cikguseven commented on src/main/java/duke/command/Command.java:
Good job on using more OOP and creating classes for the various commands!

Cikguseven commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,96 @@
+     * @throws DukeException if arguments are invalid
+     */
+    @Override
+    protected void validate(String arguments) throws DukeException {
```
Maybe the exception message can include the full format of event commands for better clarity?

Cikguseven commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,46 @@
+        String[] userInput = rawCommand.split(" ", 2);
+        String command = userInput[0];
+        String arguments = userInput.length == 2 ? userInput[1] : null;
+        switch (command) {
+        case "bye":
+            return new ByeCommand();
+        case "list":
+            return new ListCommand(arguments);
+        case "mark":
+            return new MarkCommand(arguments);
+        case "unmark":
+            return new UnmarkCommand(arguments);
+        case "delete":
+            return new DeleteCommand(arguments);
+        case "todo":
+            return new TodoCommand(arguments);
+        case "deadline":
+            return new DeadlineCommand(arguments);
+        case "event":
+            return new EventCommand(arguments);
+        default:
+            throw new DukeException("Invalid Command: " + command + " , Please Try Again...");
+        }
```
Good usage of switch case with proper indentation

nreHieW replied:
Good use of switch statements which make the code very readable!

Cikguseven reviewed (commented):
Code has proper use of OOP and is easy to read. I noticed a few minor coding standard violations
for Javadocs.

nreHieW commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,80 @@
+     *
+     * @param taskList the current list of tasks
+     */
+    public void updateFile(TaskList taskList, UI ui) {
```
This is a very neatly written function, the try catch logic is clear and explicit, making the code very readable 👍

nreHieW commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,88 @@
+     * @throws DukeException is argument is not valid
+     */
+    @Override
+    protected void validate(String arguments) throws DukeException {
```
The logic here is very clean and easy to read especially because each subsequent check is logically following the previous!

nreHieW commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,66 @@
+     *
+     * @param arguments arguments for DeleteCommand
+     */
+    public DeleteCommand(String arguments) {
```
Is there a reason you named it DeleteCommand? It makes it seem like a verb rather than a noun. Perhaps, there is no other way to name it so I guess not much can be done here. Something to take note though!

nreHieW commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,100 @@
+     * @throws DukeException if unable to list tasks
+     */
+    @Override
+    public void execute(TaskList taskList, UI ui, Storage storage) throws DukeException {
```
The logic here is also easy to follow and read 👍 , abiding by the KISS principle!

nreHieW commented on src/test/java/duke/command/DeleteCommandTest.java:
```diff
@@ -0,0 +1,70 @@
+            List<String> lines = List.of("0 todo task1");
+            assertLinesMatch(lines, Files.readAllLines(savePath.resolve("duke.txt")));
+        } catch (DukeException ignored) {
+            // Do Nothing
```
Good comment here to explain the logic behind the test!

nreHieW reviewed (approved):
Overall, the code is extremely readable and easy to understand. The coding standard was adhered to as well. LGTM 😄

B-enguin committed:
Add support for varargs message in UI.

Allow UI sendMessage() method to accept varargs.

B-enguin committed:
Merge branch 'branch-A-Varargs'

B-enguin committed:
Add support for variable height dialogBox.

Allow dialogBox to have variable size depending on message content. Add text wrapping for messages longer than dialogBox width.

B-enguin committed:
Fix GUI alignment issues.

B-enguin committed:
Update build.gradle for JavaFX dependency.

Edited build.gradle to include JavaFX dependency to include in JAR file.

B-enguin committed:
Add A-Assertions Requirements.

Add assertions to codebase to ensure that assumptions are held at various areas in the codebase. Add flag in build.gradle to enable assertions.

B-enguin committed:
Add A-CodeQuality Requirements.

Slight modifications to codebase to improve overall code quality.

B-enguin committed:
Add A-Streams Requirements.

Use stream based filtering to find task matching specified query.

B-enguin committed:
Merge pull request #1 from B-enguin/branch-A-Streams

Add A-Streams Requirements.

B-enguin committed:
Merge branch 'master' into branch-A-CodeQuality

B-enguin committed:
Merge pull request #2 from B-enguin/branch-A-CodeQuality

Add A-CodeQuality Requirements.

B-enguin committed:
Merge branch 'master' into branch-A-Assertions

B-enguin committed:
Merge pull request #3 from B-enguin/branch-A-Assertions

Add A-Assertions Requirements.

B-enguin committed:
Add CI

Add main.yml file for CI integration

B-enguin committed:
Edit test to fix TempDir bug.

B-enguin committed:
Merge remote-tracking branch 'origin/master'

B-enguin committed:
Update README.md

Add CI badge

B-enguin committed:
Add C-DetectDuplicates requirements.

TaskList will now prevent the addition of tasks that already exists, however only if the tasks details match exactly with an existing one.

B-enguin committed:
Merge branch 'branch-C-DetectDuplicates'

B-enguin committed:
Fix Storage creation bug.

Fix savefile being created as folder instead of textfile.
Minor code quality improvements.

B-enguin committed:
Add Ui Screenshot.

B-enguin committed:
Add Initial User Guide.

User Guide now shows features and how to use the application.

B-enguin committed:
Merge remote-tracking branch 'origin/master'

B-enguin committed:
Update README.md

Add chatbot name.

B-enguin committed:
Update User Guide.
Add command format to user guide.

B-enguin committed:
Merge remote-tracking branch 'origin/master'
